url4fdf79ae2c_beckchr_juel_tgz-pJ8-de_odysseus_el_samples_xml_sax_XMELFilterJ8/src/de/odysseus/el/misc/TypeConverterImpl.java

```
@@ -0,0 +1,367 @@
+			return (BigDecimal)value;
+		}
+		if (value instanceof BigInteger) {
+			return new BigDecimal((BigInteger)value);
+		}
+		if (value instanceof Number) {
+			return new BigDecimal(((Number)value).doubleValue());
+		}
+		if (value instanceof String) {
+			try {
+				return new BigDecimal((String)value);
+			} catch (NumberFormatException e) {
+				throw new ELException(LocalMessages.get("error.coerce.value", value, value.getClass(), BigDecimal.class));
+			}
+		}
+		if (value instanceof Character) {
+			return new BigDecimal((short)((Character)value).charValue());
+		}
+		throw new ELException(LocalMessages.get("error.coerce.type", value, value.getClass(), BigDecimal.class));
+	}
+
+	protected BigInteger coerceToBigInteger(Object value) {
+		if (value == null || "".equals(value)) {
+			return BigInteger.valueOf(0l);
+		}
+		if (value instanceof BigInteger) {
+			return (BigInteger)value;
+		}
+		if (value instanceof BigDecimal) {
+			return ((BigDecimal)value).toBigInteger();
+		}
+		if (value instanceof Number) {
+			return BigInteger.valueOf(((Number)value).longValue());
+		}
+		if (value instanceof String) {
+			try {
+				return new BigInteger((String)value);
+			} catch (NumberFormatException e) {
+				throw new ELException(LocalMessages.get("error.coerce.value", value, value.getClass(), BigInteger.class));
+			}
+		}
+		if (value instanceof Character) {
+			return BigInteger.valueOf((short)((Character)value).charValue());
+		}
+		throw new ELException(LocalMessages.get("error.coerce.type", value, value.getClass(), BigInteger.class));
+	}
+
+	protected Double coerceToDouble(Object value) {
+		if (value == null || "".equals(value)) {
+			return Double.valueOf(0);
+		}
+		if (value instanceof Double) {
+			return (Double)value;
+		}
+		if (value instanceof Number) {
+			return Double.valueOf(((Number)value).doubleValue());
+		}
+		if (value instanceof String) {
+			try {
+				return Double.valueOf((String)value);
+			} catch (NumberFormatException e) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)